modules/animationqt/src/animationeditordockwidgetqt.cpp

```diff
@@ -312,18 +313,47 @@ AnimationEditorDockWidgetQt::AnimationEditorDockWidgetQt(
                 [&]() { controller_.eval(controller_.getCurrentTime(), Seconds(0.0)); });
     }
 
+    {
+        // For presentation, e.g, a clicker (Key_PageUp)
+        auto* prevControlKeyframe = mainWindow_->addAction("Prev Control Keyframe");
+        prevControlKeyframe->setShortcut(Qt::Key_PageUp);  
+        prevControlKeyframe->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+        prevControlKeyframe->setToolTip("Previous Control Keyframe");
+        connect(prevControlKeyframe, &QAction::triggered, [&]() {
+            if (isKeyDoublePressed()) {
+                if (controller_.getAnimation().getTracksOfType<ControlTrack>().empty()) {
```

Comment:
Add a hasTrackOfType, to avoid allocation extra vector

Reply:
Fixed